cluster_tools/cluster_tools/schedulers/cluster_executor.py

```diff
@@ -207,15 +227,29 @@ def _completion(self, jobid, failed_early):
             logging.debug("Job completed: {}".format(jobid))
 
         preliminary_outfile_name = with_preliminary_postfix(outfile_name)
+        use_oom_exception = False
         if failed_early:
-            # If the code which should be executed on a node wasn't even
-            # started (e.g., because python isn't installed or the cluster_tools
-            # couldn't be found), no output was written to disk. We only noticed
-            # this circumstance because the whole job was marked as failed.
-            # Therefore, we don't try to deserialize pickling output.
+            # If the job failed, but didn't write the error to an output file,
+            # we handle this case separately.
+            # Typical reasons could be for this:
+            # - because python isn't installed or the cluster_tools couldn't be found
+            # - because the job was killed (e.g., by slurm due to RAM limit violations)
+            # We don't try to deserialize pickling output, because it won't exist.
             success = False
-            result = "Job submission/execution failed. Please look into the log file at {}".format(
-                self.format_log_file_path(self.cfut_dir, jobid)
+
+            reason = (  # pylint: disable=assignment-from-none
+                self.investigate_failed_job(jobid)
+            )
```

Comment:
Maybe `investigate_failed_job` could directly return an Exception itself, so that we don't need to signal the exception-type implicitly via `use_oom_exception`. Possibly also other Exceptions might be raised.

Reply:
good idea! I incorporated that. could you have a final look? :) then, I'd merge the pr.

Reply:
Sure! I probably would have returned the Exception object, already initialized with the content, and just added the outer context in the cluster-executor, but also fine as-is :+1: